src/injector/node_injector.cpp

```diff
@@ -112,18 +123,15 @@ namespace {
                         std::shared_ptr<app::Configuration> app_config,
                         std::shared_ptr<Config> genesis_config,
                         Ts &&...args) {
-    State genesis_state = STF::generateGenesisState(*genesis_config);
-    Block genesis_block = STF::genesisBlock(genesis_state);
+    AnchorState genesis_state = STF::generateGenesisState(*genesis_config);
+    AnchorBlock genesis_block = STF::genesisBlock(genesis_state);
```

Comment:
I'd like to refactor it to use only injections. AnchorState inherited on State and having ctor by genesis_config. And the same for AnchorBlock with ctor by AnchorState.